### PR TITLE
PAYM-2224 Switch to uuid5

### PR DESCRIPTION
**Pull request by Kitton**
#### :tophat: Problem
We can't update events in S3 by ourselves, we need to ask data people to remove old ones every time

#### :pushpin: Solution
We can switch to UUID5 -> event_id will be reproducible -> we can send same events for the same entities with the same event_id -> by this we will create a new version in S3 == update wrong values == we're indepent with event updates

### CODE REVIEW

File: lib/events/README.md [hunk 1]
```diff
@@ -39,10 +39,11 @@ config :ex_aws, :sns,
 
 ## Usage
 ```elixir
-Event.publish(%Event{name: "NAME_EXAMPLE", payload: %{key: "value"}, version: "1.0.0"}
+Event.publish(%Event{name: "NAME_EXAMPLE", event_id_seed: "f367d382-6452-435c-ad83-3477bd530349", payload: %{key: "value"}, version: "1.0.0"}
 ```
 Where:
 * `name` - obligatory, string, contains at least one `_`
+* `event_id_seed` - obligatory, string in UUID format, which will be used together with `name` as a seed for event_id generation. If both values will be the same - event will be updated in datalake.
```
Comment (hylkealons) [inline lib/events/README.md]: Datalake is of no concern for this library. We are not supposed to know who are clients are.

File: lib/events/adapters/aws_sns.ex [hunk 2]
```diff
@@ -38,7 +38,7 @@ defmodule ElixirTools.Events.Adapters.AwsSns do
     group = opts[:group] || Map.fetch!(config, :group)
 
     %{
-      id: UUID.uuid4(),
+      id: UUID.uuid5(event.event_id_seed, event.name),
```
Comment (hylkealons) [inline lib/events/adapters/aws_sns.ex]: wdyt about adding the version of the event as well, to ensure it's unique?

File: mix.exs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 defmodule ElixirTools.MixProject do
   use Mix.Project
 
-  @version "0.6.0"
+  @version "0.7.0"
```
Comment (hylkealons) [inline mix.exs]: never bump the version in the request itself, this can cause weird issues as a PR like this can stay open longer

File: mix.exs [hunk 4]
```diff
@@ -1,7 +1,7 @@
 defmodule ElixirTools.MixProject do
   use Mix.Project
 
-  @version "0.7.0"
+  @version "0.8.0"
```
Comment (hylkealons) [inline mix.exs]: we should not bump the version in the same PR as the functionality

Comment (Kitton) [reply mix.exs]: if PR is not long lasting and the only PR at the time it should be fine, no?

**Pull request merged**